pkg/apis/common/v1/association_test.go

```diff
@@ -89,3 +89,112 @@ func TestAssociationConfIsConfigured(t *testing.T) {
 		})
 	}
 }
+
+func TestFormatNameWithID(t *testing.T) {
+	for _, tt := range []struct {
+		name     string
+		template string
+		id       string
+		wanted   string
+	}{
+		{
+			name:     "no id",
+			template: "name%s",
+			id:       "",
+			wanted:   "name",
+		},
+		{
+			name:     "id present",
+			template: "name%s",
+			id:       "ns1-es1",
+			wanted:   "name-ns1-es1",
```

Comment:
I think using real examples here (with agent and Kibana, for example) would make the function's role even more explicit.